Mathlib/LinearAlgebra/FiniteDimensional/Lemmas.lean

```diff
@@ -238,6 +238,25 @@ theorem coe_basisOfLinearIndependentOfCardEqFinrank {ι : Type*} [Nonempty ι] [
     ⇑(basisOfLinearIndependentOfCardEqFinrank lin_ind card_eq) = b :=
   Basis.coe_mk _ _
 
+/-- In a vector space `ι → K`, a linear independent family indedex by `ι` is a basis. -/
+noncomputable def basisOfPiSpaceOfLinearIndependent {ι : Type*} [Fintype ι]
+    [Decidable (Nonempty ι)] {b : ι → (ι → K)} (hb : LinearIndependent K b) : Basis ι K (ι → K) :=
+  if hι : Nonempty ι then
+    basisOfLinearIndependentOfCardEqFinrank hb (Module.finrank_fintype_fun_eq_card K).symm
+  else
+    have : IsEmpty ι := not_nonempty_iff.mp hι
+    Basis.empty _
+
+@[simp]
+theorem coe_basisOfPiSpaceOfLinearIndependent {ι : Type*} [Fintype ι] [Decidable (Nonempty ι)]
+    {b : ι → (ι → K)} (hb : LinearIndependent K b) :
```

Comment:
```suggestion
open Classical in
@[simp]
theorem coe_basisOfPiSpaceOfLinearIndependent {ι : Type*} [Fintype ι]
    {b : ι → (ι → K)} (hb : LinearIndependent K b) :
```
For theorems I think it's better to just use `Classical` since it just removes a typeclass argument. (For defs I think it still matters, since the decidability condition actually shows up in the body of the definition.) Sorry for suggesting that change in the first place!